src/update-manager.ts

```diff
@@ -749,7 +753,106 @@ flagAuthorizationMetadata (kb?: IndexedFormula) {
   }
 
   /**
-   * This high-level function updates the local store iff the web is changed successfully.
+   * @private
+   * 
+   * This helper function constructs sparql update query from resolved arguments.
+   * 
+   * @param ds: deletions array.
+   * @param is: insertions array.
+   * @param bnodes_context: Additional context to uniquely identify any blanknodes.
+   */
+  constructSparqlUpdateQuery(
+    ds: ReadonlyArray<Statement>,
+    is: ReadonlyArray<Statement>,
+    bnodes_context,
+  ): string {
+    var whereClause = this.contextWhere(bnodes_context)
+    var query = ''
+    if (whereClause.length) { // Is there a WHERE clause?
+      if (ds.length) {
+        query += 'DELETE { '
+        for (let i = 0; i < ds.length; i++) {
+          query += this.anonymizeNT(ds[i]) + '\n'
+        }
+        query += ' }\n'
+      }
+      if (is.length) {
+        query += 'INSERT { '
+        for (let i = 0; i < is.length; i++) {
+          query += this.anonymizeNT(is[i]) + '\n'
+        }
+        query += ' }\n'
+      }
+      query += whereClause
+    } else { // no where clause
+      if (ds.length) {
+        query += 'DELETE DATA { '
+        for (let i = 0; i < ds.length; i++) {
+          query += this.anonymizeNT(ds[i]) + '\n'
+        }
+        query += ' } \n'
+      }
+      if (is.length) {
+        if (ds.length) query += ' ; '
+        query += 'INSERT DATA { '
+        for (let i = 0; i < is.length; i++) {
+          query += this.nTriples(is[i]) + '\n'
+        }
+        query += ' }\n'
+      }
+    }
+    return query;
+  }
+
+  /**
+   * @private
+   * 
+   * This helper function constructs n3-patch query from resolved arguments.
+   * 
+   * @param ds: deletions array.
+   * @param is: insertions array.
+   * @param bnodes_context: Additional context to uniquely identify any blanknodes.
+   */
+  constructN3PatchQuery(
```

Comment:
I think it would be good to unit test this function properly to cover it in all it's complexity. The current test is more like an integration test (which is fine!). You could add some tests for this method specifically, call it with different intputs and assert that the correct string is constructed.

Reply:
Current test file is just an adaptation of corresponding sparql-update's test file. Indeed i would want to add proper unit tests for the fn. But as of now i am not so familiar with these magic test harnesses of js ecosystem with their meta language. I would be learning jest, mocha soon though.

Along with that, i need to add test cases for n3-patch implementation in manas.

Hence i could address this in a separate PR in very near future. For now i tested with solidos against CSS manually. It "seems" to work as expected.

Reply:
Added more tests, that tests for delete statements and complex cases with blank nodes, where they must be anonymized, and a conditions formula needs to be specified to resolve them back. @angelo-v

Reply:
thank you!